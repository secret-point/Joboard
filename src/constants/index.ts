export const IN_PROGRESS = "IN_PROGRESS";
export const PENDING = "PENDING";
export const COMPLETED = "COMPLETED";

export const MAX_MINUTES_FOR_HEARTBEAT = 60;

export const IGNORE_PAGE_TO_STORE_LOCAL: string[] = ["wotc-complete"];

export const CS_DOMAIN_LIST = [
    "https://beta-us.devo.jobsatamazon.hvh.a2z.com",
    "https://gamma-us.devo.jobsatamazon.hvh.a2z.com",
    "https://hiring.amazon.com",
];