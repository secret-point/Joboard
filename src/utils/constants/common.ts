export const HVH_LOCALE = 'hvh-locale';
