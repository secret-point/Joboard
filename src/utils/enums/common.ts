import { PAGE_ROUTES } from "../../components/pageRoutes";

const { NHE } = PAGE_ROUTES;

export enum QUERY_PARAMETER_NAME {
    APPLICATION_ID = 'applicationId',
    JOB_ID = 'jobId',
    SCHEDULE_ID = 'scheduleId',
    REQUISITION_ID = 'requisitionId'
}

export enum APPLICATION_STEPS {
    SELECT_JOB = 'Select Job',
    COMPLETE_REQUIRED_DOCUMENTS = 'Complete required documents',
    SCHEDULE_PRE_HIRE_APPOINTMENT = 'Schedule pre-hire appointment',
    COMPLETE_AN_ASSESSMENT = 'Complete an assessment (~15 min)'
}

export enum SCHEDULE_FILTER_TYPE {
    PAY_RATE = "PAY_RATE",
    HOURS_DESC = "HOURS_DESC",
    HOURS_ASC = "HOURS_ASC",
    DEFAULT = "DEFAULT"
}

export enum DESIRED_WORK_HOURS {
    TEN = "10",
    TWENTY = "20",
    THIRTY = "30",
    FORTY = "40"
}

export enum DAYS_OF_WEEK {
    MONDAY= "MONDAY",
    TUESDAY = "TUESDAY",
    WEDNESDAY = "WEDNESDAY",
    THURSDAY = "THURSDAY",
    FRIDAY = "FRIDAY",
    SATURDAY = "SATURDAY",
    SUNDAY = "SUNDAY"
}

export enum FCRA_DISCLOSURE_TYPE {
    ACCEPT = "Accept",
    DECLINE = "Decline"
}

export enum CRIMINAL_CONVICTION_STATUS {
    ACCEPT = "ACCEPT",
    DECLINE = "DECLINE"
}

export enum NON_FCRA_STATES {
    ALL_STATE = "ALL STATES",
    WASHINGTON = "WASHINGTON",
    NEW_YORK = "NEW YORK",
    MINNESOTA = "MINNESOTA",
    CALIFORNIA = "CALIFORNIA"
}

export enum BGC_STEPS {
    FCRA = "FCRA",
    NON_FCRA = "NON_FCRA",
    ADDITIONAL_BGC = "ADDITIONAL_BGC"
}

export enum SELF_IDENTIFICATION_STEPS {
    EQUAL_OPPORTUNITY = "EQUAL_OPPORTUNITY_FORM",
    VETERAN_FORM = "VETERAN_FORM",
    DISABILITY_FORM = "DISABILITY_FORM"
}

export type INFO_CARD_STEPS = BGC_STEPS | SELF_IDENTIFICATION_STEPS;

export enum INFO_CARD_STEP_STATUS {
    LOCKED = 'LOCKED',
    ACTIVE = 'ACTIVE',
    COMPLETED = 'COMPLETED'
}

export enum WORKFLOW_STEP_NAME {
    JOB_OPPORTUNITIES = 'job-opportunities',
    ASSESSMENT_CONSENT = 'assessment-consent',
    CONTINGENT_OFFER = 'contingent-offer',
    THANK_YOU = 'thank-you',
    SUPPLEMENTARY_SUCCESS = 'supplementary-success',
    BGC = 'bgc',
    AMAZON_REJECTS = 'amazon-rejects',
    AMAZON_WITHDRAWS = 'amazon-withdraws',
    CANDIDATE_WITHDRAWS = 'candidate-withdraws',
    REVIEW_SUBMIT = 'review-submit',
    WOTC = 'wotc',
    SELF_IDENTIFICATION = 'self-identification',
    NHE = 'nhe',
    WORKFLOW_FAILED = 'workflow-failed',
    REHIRE_ELIGIBILITY_STATUS = 'rehire-eligibility-status',
    CAN_NOT_OFFER_JOB = 'can-not-offer-job',
    CALI_DISCLOSURE = 'cali-disclosure',
    ASSESSMENT_NOT_ELIGIBLE = 'assessment-not-eligible'
}

export enum UPDATE_APPLICATION_API_TYPE {
    JOB_CONFIRM = 'job-confirm',
    REVIEW_SUBMIT = 'review-submit',
    CONTINGENT_OFFER = 'contingent-offer',
    BGC = 'bgc',
    NON_FCRA_BGC = 'non-fcra',
    FCRA_BGC = 'fcra',
    ADDITIONAL_BGC = 'additional-bgc-info',
    CALI_DISCLOSURE = 'cali-disclosure',
    EQUAL_OPPORTUNITY_FORM = 'equal-opportunity-form',
    VETERAN_STATUS_FORM = 'veteran-status-form',
    DISABILITY_FORM = 'disability-form',
    SELF_IDENTIFICATION = 'self-identification',
    NHE = 'nhe',
    UPDATE_SHIFT = 'update-shift',
    UPDATE_SCHEDULE = 'update-schedule',
    CANCEL_SHIFT = 'cancel-shift',
    CANCEL_SCHEDULE = 'cancel-schedule',
    THANK_YOU = 'thank-you',
    JOB_PREFERENCES = 'job-preferences',
    NHE_PREFERENCES = 'nhe=preferences',
    DEFAULT = 'Update type is missing'
}

export enum BACKGROUND_AGENT {
    FADV = 'FADV',
    ACCURATE = 'ACCURATE'
}

export enum APPLICATION_STATE {
    ASSESSMENT_PENDING = "ASSESSMENT_PENDING",
    SHIFT_SELECTION_PENDING = "SHIFT_SELECTION_PENDING",
    CONDITIONAL_INTENT_PENDING = "CONDITIONAL_INTENT_PENDING",
    BGC_CONSENT_AND_INFO_PENDING = "BGC_CONSENT_AND_INFO_PENDING",
    PERSONAL_DETAILS_PENDING = "PERSONAL_DETAILS_PENDING",
    LOCAL_BGC_CONSENT_PENDING = "LOCAL_BGC_CONSENT_PENDING",
    APPT_1_SCHEDULING = "APPT_1_SCHEDULING",
    DIVERSITY_PENDING = "DIVERSITY_PENDING",
    REVIEW_AND_SUBMIT_PENDING = "REVIEW_AND_SUBMIT_PENDING",
    EVALUATION_PENDING = "EVALUATION_PENDING",
    APPT_1_PENDING = "APPT_1_PENDING",
    CONTINGENCIES_PENDING = "CONTINGENCIES_PENDING",
    SCHEDULE_SELECTION_PENDING = "SCHEDULE_SELECTION_PENDING",
    START_DATE_CONFIRMATION_PENDING = "START_DATE_CONFIRMATION_PENDING",
    READY_TO_HIRE = "READY_TO_HIRE",
    HIRE_COMPLETE = "HIRE_COMPLETE",
    WITHDRAWN = "WITHDRAWN",
    REJECTED_BY_AMAZON = "REJECTED_BY_AMAZON"
}

export enum PROXY_APPLICATION_STATE {
    APPLICATION_CREATED = "APPLICATION_CREATED",
    ASSESSMENT_PASSED = "ASSESSMENT_PASSED",
    JOB_SELECTED = "JOB_SELECTED",
    CONTINGENT_OFFER_ACCEPTED = "CONTINGENT_OFFER_ACCEPTED",
    FCRA_CONSENT_SAVED = "FCRA_CONSENT_SAVED",
    NON_FCRA_CONSENT_SAVED = "NON_FCRA_CONSENT_SAVED",
    ADDITIONAL_BACKGROUND_INFO_SAVED = "ADDITIONAL_BACKGROUND_INFO_SAVED",
    CALIFORNIA_DISCLOSURE_SAVED = "CALIFORNIA_DISCLOSURE_SAVED",
    JOB_REFERRAL_SAVED = "JOB_REFERRAL_SAVED",
    PRE_HIRE_APPOINTMENT_SCHEDULED = "PRE_HIRE_APPOINTMENT_SCHEDULED",
    NHE_PREFERENCE_SELECTED = "NHE_PREFERENCE_SELECTED",
    NHE_APPOINTMENT_SELECTED = "NHE_APPOINTMENT_SELECTED",
    APPLICATION_SUBMITTED = "APPLICATION_SUBMITTED",
    APPLICATION_WITHDRAWN = "APPLICATION_WITHDRAWN",
    WOTC_COMPLETED = "WOTC_COMPLETED",
    WOTC_INITIATED = "WOTC_INITIATED",
    EQUAL_OPPORTUNITY_FORM_SAVED = "EQUAL_OPPORTUNITY_FORM_SAVED",
    VETERAN_STATUS_SAVED = "VETERAN_STATUS_SAVED",
    SELF_IDENTIFICATION_COMPLETED = "SELF_IDENTIFICATION_COMPLETED",
}

export enum PROXY_ERROR_MESSAGE {
    //application
    BGC_VALIDATION_FAILED = "Additional background information is not valid, Please update all required fields",
    MISSING_REFERRAL_INFORMATION = "Missing referral information",
    FAILED_UPDATE_APPLICATION = "Failed to update the application",
    NO_ACCESS_TO_APPLICATION = "you can not access this application",
    FAILED_TERMINATE_APPLICATION = "Failed to terminate application",
    FAILED_UPDATE_WOTC = "Failed to update WOTC status",
    FAILED_UPDATE_STEP = "Failed to updating the step name",
    CAN_NOT_CREATE_APPLICATION = "you can not create application",
    APPLICATION_ID_REQUIRED = "applicationId is required to get application data",
    DUPLICATE_SSN = 'The SSN or National Identity Number you entered is already associated with another email address in our system. If you have previously signed up with a different email address, please [Log Out](/logout) and try logging in with that email, using the "Forgot your personal pin?" link if needed. If you need help, please contact us via email or chat at [www.amazon.com/applicationhelp](https://www.amazondelivers.jobs/contactus)',
    MAX_SSN_EDITS = 'Update to application failed because you have reached the maximum number of changes to your National ID. If you have previously signed up with a different email address, please [Log Out](/logout) and try logging in with that email, using the "Forgot your personal pin?" link if needed. If you need help, please contact us via email or chat at [www.amazon.com/applicationhelp](https://www.amazondelivers.jobs/contactus)',
    INVALID_SSN = "Update to application failed because provided National Identity Number is invalid. Please enter correct and complete National Identity Number before proceeding.",
    MISMATCH_SIGNATURES = "eSignatures do not match. Please use the same text for each eSignature.",
    //requisition
    FETCH_REQUISITION_ERROR = "Unable to fetch requisition.",
    REQUISITION_INFO_ERROR = "Unable to fetch requisition information.",
    FETCH_CHILD_REQUISITION_ERROR = "Unable to fetch child requisition.",
    JOB_DES_ERROR = "Unable to fetch job description.",
    FETCH_TIME_SLOT_ERROR = "Something went wrong! Please refresh the page or try again later",
    ALL_SHIFT_FILLED = "All Shifts Filled",
    FETCH_SHIFTS_ERROR = "Something went wrong while fetching jobs! Please refresh the page or try again later",
    FETCH_JOB_ERROR = "Unable to fetch job.",
    JOB_INFO_ERROR = "Unable to fetch job information.",
    FETCH_HCR_ERROR = "Unable to fetch head count request.",
    FETCH_SHIFT_PREF_ERROR = "Unable to fetch shift preferences.",
    SHIFTS_BAD_REQUEST = "Failed to retrieve jobs at the moment. Please try again after some time.",
    SCHEDULE_NOT_AVAILABLE = "The schedule you have selected is no longer available, please select another schedule.",
    REFERRER_NOT_FOUND = "We couldn't validate your referrer's login ID, please verify with your referrer and try again.",
    REFERRAL_VALIDATION_BLOCKED = "We couldn't validate your referrer's login ID, please chat with us at www.amazon.com/applicationhelp for assistance.",
    REFERRAL_VALIDATION_ERROR = "We couldn't validate your referrer's login ID, please try again or refresh the page.",
}

export enum CREATE_APPLICATION_ERROR_CODE {
    MISSING_REQUIRED_FIELDS = "MISSING_REQUIRED_FIELDS",
    CANDIDATE_NOT_FOUND = "CANDIDATE_NOT_FOUND",
    CANDIDATE_IS_INACTIVE = "CANDIDATE_IS_INACTIVE",
    APPLICATION_ALREADY_EXIST = "APPLICATION_ALREADY_EXIST",
    VERSION_MISMATCH = "VERSION_MISMATCH",
    INTERNAL_SERVER_ERROR = "INTERNAL_SERVER_ERROR",
}

export enum UPDATE_APPLICATION_ERROR_CODE {
    MISSING_REQUIRED_FIELDS = "MISSING_REQUIRED_FIELDS",
    APPLICATION_ID_DOES_NOT_MATCH = "APPLICATION_ID_DOES_NOT_MATCH",
    APPLICATION_NOT_FOUND = "APPLICATION_NOT_FOUND",
    VERSION_MISMATCH = "VERSION_MISMATCH",
    INTERNAL_SERVER_ERROR = "INTERNAL_SERVER_ERROR",
    CANDIDATE_NOT_AUTHORIZED = "CANDIDATE_NOT_AUTHORIZED",
    CONTINGENT_OFFER_NOT_ACCEPTED = "CONTINGENT_OFFER_NOT_ACCEPTED",
    CONTINGENT_OFFER_ACCEPTED_TIME_IS_NULL = "CONTINGENT_OFFER_ACCEPTED_TIME_IS_NULL",
    SIGN_FCRA_ACKNOWLEDGEMENT = "SIGN_FCRA_ACKNOWLEDGEMENT",
    SIGN_FCRA_REQUEST = "SIGN_FCRA_REQUEST",
    SIGN_NON_FCRA_ACKNOWLEDGEMENT = "SIGN_NON_FCRA_ACKNOWLEDGEMENT",
    SIGN_NON_FCRA_STATE_NOTICE = "SIGN_NON_FCRA_STATE_NOTICE",
    SELECTED_SCHEDULE_NOT_AVAILABLE = "SELECTED_SCHEDULE_NOT_AVAILABLE",
    MISMATCH_SIGNATURES = "MISMATCH_SIGNATURES",
    MAX_SSN_EDITS = "MAX_SSN_EDITS",
    INVALID_SSN = "INVALID_SSN",
    TYPE_OF_REQUEST_MISSING = "TYPE_OF_REQUEST_MISSING",
    TYPE_NOT_MATCH_WITH_REQUEST = "TYPE_NOT_MATCH_WITH_REQUEST",
    DOB_MISSING_IN_REQUEST = "DOB_MISSING_IN_REQUEST",
    PROVIDE_DOB = "PLEASE_PROVIDE_DOB",
    APPLICATION_ID_MISSING = "APPLICATION_ID_MISSING",
    GOVERNMENT_TYPE_MISSING_IN_REQUEST = "GOVERNMENT_TYPE_MISSING_IN_REQUEST",
    ID_NUMBER_MISSING_IN_REQUEST = "ID_NUMBER_MISSING_IN_REQUEST",
    HAS_CRIMINAL_IN_SEVEN_YEARS_MISSING = "HAS_CRIMINAL_IN_SEVEN_YEARS_MISSING",
    ADDRESS_LINE1_MISSING = "ADDRESS_LINE1_MISSING",
    CITY_IS_MISSING_IN_ADDRESS = "CITY_IS_MISSING_IN_ADDRESS",
    STATE_IS_MISSING_IN_ADDRESS = "STATE_IS_MISSING_IN_ADDRESS",
    ZIPCODE_MISSING_IN_ADDRESS = "ZIPCODE_MISSING_IN_ADDRESS",
    COUNTRY_CODE_MISSING_IN_ADDRESS = "COUNTRY_CODE_MISSING_IN_ADDRESS",
    INVALID_PREVIOUS_LEGAL_NAME = "INVALID_PREVIOUS_LEGAL_NAME",
    BGC_VALIDATION_FAILED = 'VALIDATION_FAILED',
    NHE_APT_MISSING_IN_REQUEST = "NHE_APT_MISSING_IN_REQUEST",
    REFERRAL_INFO_MISSING = "REFERRAL_INFO_MISSING",
    ETHNICITY_IS_MISSING = "ETHNICITY_IS_MISSING",
    GENDER_IS_MISSING = "GENDER_IS_MISSING",
    MILITARY_SPOUSE_MISSING = "MILITARY_SPOUSE_MISSING",
    PROTECTED_VETERAN_MISSING = "PROTECTED_VETERAN_MISSING",
    VETERAN_STATUS_MISSING = "VETERAN_STATUS_MISSING",
    DISABILITY_IS_MISSING = "DISABILITY_IS_MISSING",
    CALI_DISCLOSURE_IS_MISSING = "CALI_DISCLOSURE_IS_MISSING",
    DUPLICATE_SSN = "DUPLICATE_SSN",
    CAN_NOT_CREATE_APPLICATION = "CAN_NOT_CREATE_APPLICATION",
    UNABLE_CREATE_APPLICATION = "UNABLE_CREATE_APPLICATION",
    CANDIDATE_ALREADY_HIRED = "CANDIDATE_ALREADY_HIRED",
    FAILED_UPDATE_APPLICATION = "FAILED_UPDATE_APPLICATION",
    SELECTED_JOB_NOT_AVAILABLE = "SELECTED_JOB_NOT_AVAILABLE",
    PREVIOUS_USED_NAME_FORMAT_INCORRECT = "PREVIOUS_USED_NAME_FORMAT_INCORRECT"
}

export enum GET_APPLICATION_ERROR_CODE {
    MISSING_REQUIRED_FIELDS = "MISSING_REQUIRED_FIELDS",
    CANDIDATE_NOT_AUTHORIZED = "CANDIDATE_NOT_AUTHORIZED",
    APPLICATION_NOT_FOUND = "APPLICATION_NOT_FOUND",
    INTERNAL_SERVER_ERROR = "INTERNAL_SERVER_ERROR",
}

export enum UPDATE_WORKFLOW_NAME_ERROR_CODE {
    MISSING_REQUIRED_FIELDS = "MISSING_REQUIRED_FIELDS",
    APPLICATION_NOT_FOUND = "APPLICATION_NOT_FOUND",
    INTERNAL_SERVER_ERROR = "INTERNAL_SERVER_ERROR",
}

export enum GET_JOB_INFO_ERROR_CODE {
    FETCH_JOB_ERROR = "FETCH_JOB_ERROR"
}

export enum GET_SCHEDULE_LIST_BY_JOB_ID_ERROR_CODE {
    ALL_SHIFT_FILLED = " ALL_SHIFT_FILLED",
    FETCH_SHIFTS_ERROR = "FETCH_SHIFTS_ERROR",
    SHIFTS_BAD_REQUEST = "SHIFTS_BAD_REQUEST",
    NO_SCHEDULE_FOUND = "NO_SCHEDULE_FOUND"
}

export enum GET_SCHEDULE_DETAIL_ERROR_CODE {
    FETCH_HCR_ERROR = "FETCH_HCR_ERROR",
}

export enum GET_NHE_TIME_SLOT_LIST_ERROR_CODE {
    FETCH_TIME_SLOT_ERROR = "FETCH_TIME_SLOT_ERROR",
    ALL_SHIFT_FILLED = "ALL_SHIFT_FILLED",
    SHIFTS_BAD_REQUEST = "SHIFTS_BAD_REQUEST",
    FETCH_SHIFTS_ERROR = "FETCH_SHIFTS_ERROR",
    FETCH_SHIFT_PREF_ERROR = "FETCH_SHIFT_PREF_ERROR"
}

export enum GET_CANDIDATE_ERROR_CODE {
    FAILED_GET_CANDIDATE = "FAILED_GET_CANDIDATE",
    NO_ACCESS_TO_APPLICATION = "NO_ACCESS_TO_APPLICATION"
}

export enum UPDATE_WOTC_STATUS_ERROR_CODE {
    FAILED_UPDATE_WOTC = "FAILED_UPDATE_WOTC",
}

export enum VALIDATE_AMAZON_LOGIN_ID_ERROR_CODE {
    REFERRAL_INPUT_EMPTY = "REFERRAL_INPUT_EMPTY",
    REFERRAL_INPUT_FORMAT_INVALID = "REFERRAL_INPUT_FORMAT_INVALID",
    REFERRER_NOT_FOUND = "REFERRER_NOT_FOUND",
    REFERRAL_VALIDATION_BLOCKED = "REFERRAL_VALIDATION_BLOCKED",
    REFERRAL_VALIDATION_ERROR = "REFERRAL_VALIDATION_ERROR",
}

export enum WORKFLOW_ERROR_CODE {
    NOT_REHIRE_ELIGIBLE = "not-rehire-eligible",
    NOT_REHIRE_ELIGIBLE_365_DAYS = "365-days-not-rehire-eligible",
    ACTIVE = "active",
    SEASONAL_ONLY = "seasonal-only"
}

export enum FEATURE_FLAG {
    PREFERRED_NAME = "PREFERRED_NAME",
    PREVENT_DUPLICATE_SSN = "PREVENT_DUPLICATE_SSN",
    SUPPRESS_QUESTIONS_IF_REHIRE_ELIGIBLE = "SUPPRESS_QUESTIONS_IF_REHIRE_ELIGIBLE",
    STENCIL_COLOR_OVERRIDE = "STENCIL_COLOR_OVERRIDE",
    CS_AUTH = "CS_AUTH",
    UNIFIED_DOMAIN = "UNIFIED_DOMAIN",
    SELF_ID_PLACEMENT = "SELF_ID_PLACEMENT",
    CANDIDATE_DASHBOARD = "CANDIDATE_DASHBOARD",
    HOOK = "HOOK",
    NO_SSN_CHECKBOX = "NO_SSN_CHECKBOX",
    MLS = "MLS",
    REFERRAL_VALIDATION = "REFERRAL_VALIDATION",
    BACKEND_HARD_RESERVE_SCHEDULE = "BACKEND_HARD_RESERVE_SCHEDULE",
}

export enum BB_UI_VERSION {
    BB_UI_OLD = "bb-ui-v1",
    BB_UI_NEW = "bb-ui-v2",
}

export enum BGC_VENDOR_TYPE {
    FADV = "FADV",
    ACCURATE = "ACCURATE"
}
