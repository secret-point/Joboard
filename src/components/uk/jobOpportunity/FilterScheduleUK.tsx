import React, { useState } from "react";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { Label, Text } from "@amzn/stencil-react-components/text";
import { translate as t } from "../../../utils/translator";
import { ScheduleStateFilters } from "../../../utils/types/common";
import FilterSchedule from "../../common/jobOpportunity/FilterSchedule";
import { Checkbox, InputWrapper, LabelPosition, InputFooter } from "@amzn/stencil-react-components/form";
import { PopupDatePicker } from "@amzn/stencil-react-components/date-picker";
import { isDateGreaterThanToday } from "../../../utils/helper";
import { boundUpdateScheduleFilters } from "../../../actions/ScheduleActions/boundScheduleActions";

interface FilterScheduleProps {
  filters: ScheduleStateFilters;
  supportedCities: string[];
  scheduleCount: number;
}

const FilterScheduleUK = ( props: FilterScheduleProps ) => {

  const { filters, supportedCities, scheduleCount } = props;
  const { city } = filters;
  const [startEndDate, setStartEndDate] = useState<string|undefined>(undefined);
  const [startBeginDate, setStartBeginDate] = useState<string|undefined>(undefined);
  const isStartDateValid = startBeginDate ? isDateGreaterThanToday(startBeginDate) : true;
  const isEndDateValid = startEndDate ? isDateGreaterThanToday(startEndDate) : true;

  const updateStartDate = ( key: string ) => {
    const newFilters = { ...filters };
    newFilters.startDateBegin = key;
    setStartBeginDate(key);

    isStartDateValid && boundUpdateScheduleFilters(newFilters);
  };

  const updateEndDate = ( key: string ) => {
    const newFilters = { ...filters };
    newFilters.startDateEnd = key;
    setStartEndDate(key);

    isEndDateValid && boundUpdateScheduleFilters(newFilters);
  };

  const updateCityCheckbox = (city: string) => {
    const newFilters = { ...filters };
    const currentCity = newFilters.city || [];
    if (currentCity.indexOf(city) > -1) {
      currentCity.splice(currentCity.indexOf(city), 1);
    } else {
      currentCity.push(city);
    }
    newFilters.city = currentCity;

    boundUpdateScheduleFilters(newFilters);
  };

  return (
    <Col gridGap={20}>
      <Col gridGap={10} width="100%" padding={{ right: "S300" }}>
        <Text textAlign="center">
          {t("BB-FilterSchedule-schedule-found-counter-text", `${scheduleCount} Schedule Found`, { scheduleCount: scheduleCount })}
        </Text>
        <Col gridGap={8}>
          <Text>Cities</Text>
          <Row gridGap={10}>
            {
              supportedCities?.map(cityItem => (
                <Row alignItems="end" key={cityItem} gridGap={8}>
                  <InputWrapper
                    labelText={""}
                    id={`checkbox-city-${cityItem}`}
                    labelPosition={LabelPosition.Top}
                  >
                    {inputProps => (
                      <Checkbox
                        name="city" {...inputProps}
                        checked={city && city.indexOf(cityItem) > -1}
                        onChange={() => updateCityCheckbox(cityItem)}
                      />
                    )}
                  </InputWrapper>
                  <Label htmlFor={`checkbox-city-${cityItem}`}>{cityItem}</Label>
                </Row>
              ))
            }
          </Row>
        </Col>
        <Col gridGap={8} padding={{ top: "S300" }}>
          <Text>{t("BB-FilterScheduleUk-start-date-between-title", "Start date between")}</Text>
          <Col padding={{ top: "S200" }}>
            <PopupDatePicker
              inputProps={{ width: "100%", error: !isStartDateValid }}
              isDateDisabled={value => !isDateGreaterThanToday(value)}
              value={startBeginDate}
              onChange={updateStartDate}
              id="startDateBegin"
            />
            {!isStartDateValid && (
              <InputFooter error={!isStartDateValid} id="startDateBeginErrorMessage">
                {t("BB-FilterScheduleUk-start-date-between-error-text", "Please select a valid date")}
              </InputFooter>
            )}
          </Col>
          <Col padding={{ top: "S200" }}>
            <PopupDatePicker
              inputProps={{ width: "100%", error: !isEndDateValid }}
              isDateDisabled={value => !isDateGreaterThanToday(value)}
              value={startEndDate}
              onChange={updateEndDate}
              id="startDateEnd"
            />
            {!isEndDateValid && (
              <InputFooter error={!isEndDateValid} id="startDateEndErrorMessage">
                {t("BB-FilterScheduleUk-end-date-between-error-text", "Please select a valid date")}
              </InputFooter>
            )}
          </Col>
        </Col>
      </Col>
      <FilterSchedule filters={filters} displayMilitaryTime />
    </Col>
  );
};

export default FilterScheduleUK;
