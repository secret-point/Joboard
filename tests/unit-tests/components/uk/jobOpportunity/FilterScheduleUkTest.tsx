import React from "react";
import { shallow } from "enzyme";
import { dayHoursFilterValues, initScheduleStateFilters } from "../../../../../src/utils/constants/common";
import FilterScheduleUK from "../../../../../src/components/uk/jobOpportunity/FilterScheduleUK";
import { mountWithStencil } from "@amzn/stencil-react-components/tests";

describe("ShiftPreference Card", () => {

  it("should match snapshot", () => {
    const shallowWrapper = shallow(
      <FilterScheduleUK
        filters={{
          ...initScheduleStateFilters,
          daysHoursFilter: dayHoursFilterValues
        }}
        scheduleCount={5}
        supportedCities={["test1", "test2"]}
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });

  it("should show error message on invalid date on startDate", () => {
    let wrapper = mountWithStencil(
      <FilterScheduleUK
        filters={{
          ...initScheduleStateFilters,
          daysHoursFilter: dayHoursFilterValues
        }}
        scheduleCount={5}
        supportedCities={["test1", "test2"]}
      />);
    const dateInput = wrapper.find("input#startDateBegin");
    dateInput.simulate("change", { target: { value: "20/20/20000" } });

    wrapper = wrapper.update();
    const errorMessage = wrapper.find("InputFooter#startDateBeginErrorMessage");
    expect(errorMessage.text()).toContain("Please select a valid date");
  });

  it("should show error message on invalid date on startEndDate", () => {
    let wrapper = mountWithStencil(
      <FilterScheduleUK
        filters={{
          ...initScheduleStateFilters,
          daysHoursFilter: dayHoursFilterValues
        }}
        scheduleCount={5}
        supportedCities={["test1", "test2"]}
      />);
    const dateInput = wrapper.find("input#startDateEnd");
    dateInput.simulate("change", { target: { value: "20/20/20000" } });

    wrapper = wrapper.update();
    const errorMessage = wrapper.find("InputFooter#startDateEndErrorMessage");
    expect(errorMessage.text()).toContain("Please select a valid date");
  });
});