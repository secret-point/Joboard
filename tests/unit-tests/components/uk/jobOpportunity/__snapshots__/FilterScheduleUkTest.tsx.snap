// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShiftPreference Card should match snapshot 1`] = `
<Col
  gridGap={20}
>
  <Col
    gridGap={10}
    padding={
      Object {
        "right": "S300",
      }
    }
    width="100%"
  >
    <Text
      font="primary"
      fontSize="T300"
      hyphens="none"
      textAlign="center"
    >
      5 Schedule Found
    </Text>
    <Col
      gridGap={8}
    >
      <Text
        font="primary"
        fontSize="T300"
        hyphens="none"
      >
        Cities
      </Text>
      <Row
        gridGap={10}
      >
        <Row
          alignItems="end"
          gridGap={8}
          key="test1"
        >
          <InputWrapper
            error={false}
            id="checkbox-city-test1"
            labelPosition="top"
            labelText=""
            required={false}
            shouldValidateNonInclusiveTerms={false}
            warning={false}
          >
            <Component />
          </InputWrapper>
          <Label
            color="neutral90"
            font="primary"
            fontSize="T200"
            fontWeight="regular"
            htmlFor="checkbox-city-test1"
            hyphens="none"
          >
            test1
          </Label>
        </Row>
        <Row
          alignItems="end"
          gridGap={8}
          key="test2"
        >
          <InputWrapper
            error={false}
            id="checkbox-city-test2"
            labelPosition="top"
            labelText=""
            required={false}
            shouldValidateNonInclusiveTerms={false}
            warning={false}
          >
            <Component />
          </InputWrapper>
          <Label
            color="neutral90"
            font="primary"
            fontSize="T200"
            fontWeight="regular"
            htmlFor="checkbox-city-test2"
            hyphens="none"
          >
            test2
          </Label>
        </Row>
      </Row>
    </Col>
    <Col
      gridGap={8}
      padding={
        Object {
          "top": "S300",
        }
      }
    >
      <Text
        font="primary"
        fontSize="T300"
        hyphens="none"
      >
        Start date between
      </Text>
      <Col
        padding={
          Object {
            "top": "S200",
          }
        }
      >
        <PopupDatePicker
          calendarButtonAltText="Calendar"
          collapsedCloseButtonAltText="Close date picker"
          collapsedTitleText="Select a date"
          id="startDateBegin"
          inputProps={
            Object {
              "error": false,
              "width": "100%",
            }
          }
          isDateDisabled={[Function]}
          onChange={[Function]}
          popoverProps={
            Object {
              "position": "bottom-trailing",
            }
          }
          renderPopupContent={[Function]}
        />
      </Col>
      <Col
        padding={
          Object {
            "top": "S200",
          }
        }
      >
        <PopupDatePicker
          calendarButtonAltText="Calendar"
          collapsedCloseButtonAltText="Close date picker"
          collapsedTitleText="Select a date"
          id="startDateEnd"
          inputProps={
            Object {
              "error": false,
              "width": "100%",
            }
          }
          isDateDisabled={[Function]}
          onChange={[Function]}
          popoverProps={
            Object {
              "position": "bottom-trailing",
            }
          }
          renderPopupContent={[Function]}
        />
      </Col>
    </Col>
  </Col>
  <FilterSchedule
    displayMilitaryTime={true}
    filters={
      Object {
        "daysHoursFilter": Array [
          Object {
            "day": "MONDAY",
            "dayTranslationKey": "BB-DayName-Monday",
            "endTime": "23:59",
            "isActive": true,
            "startTime": "00:00",
          },
          Object {
            "day": "TUESDAY",
            "dayTranslationKey": "BB-DayName-Tuesday",
            "endTime": "23:59",
            "isActive": true,
            "startTime": "00:00",
          },
          Object {
            "day": "WEDNESDAY",
            "dayTranslationKey": "BB-DayName-Wednesday",
            "endTime": "23:59",
            "isActive": true,
            "startTime": "00:00",
          },
          Object {
            "day": "THURSDAY",
            "dayTranslationKey": "BB-DayName-Thursday",
            "endTime": "23:59",
            "isActive": true,
            "startTime": "00:00",
          },
          Object {
            "day": "FRIDAY",
            "dayTranslationKey": "BB-DayName-Friday",
            "endTime": "23:59",
            "isActive": true,
            "startTime": "00:00",
          },
          Object {
            "day": "SATURDAY",
            "dayTranslationKey": "BB-DayName-Saturday",
            "endTime": "23:59",
            "isActive": true,
            "startTime": "00:00",
          },
          Object {
            "day": "SUNDAY",
            "dayTranslationKey": "BB-DayName-Sunday",
            "endTime": "23:59",
            "isActive": true,
            "startTime": "00:00",
          },
        ],
        "maxHoursPerWeek": "40",
        "sortKey": "DEFAULT",
      }
    }
  />
</Col>
`;
